### PR TITLE
Move more scripts to scripts/

### PR DESCRIPTION
**Pull request by sleepinyourhat**
@iftenney, mind taking a look?

### CODE REVIEW

**Comment by sleepinyourhat** [comment]
Resolves #606.

File: demo.with_docker.sh [hunk 1]
```diff
@@ -40,7 +40,6 @@ COMMAND+=( -o "exp_name=jiant-demo" )
 # Run demo.conf in the docker container.
 sudo docker run --runtime=nvidia --rm -v "$TEMP_DIR:/nfs/jsalt" \
   -v "$JIANT_PATH:/share/jiant" \
-  -e "NFS_PROJECT_PREFIX=/nfs/jsalt/exp" \
```
Comment (iftenney) [inline demo.with_docker.sh]: Does this assume #615 ?

File: scripts/demo.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
-# Quick-start: run this
+# Quick-start: run this from the base jiant/ directory
```
Comment (iftenney) [inline scripts/demo.sh]: Optional, but I like to add a `pushd` to automatically jump to this directory - just smooths out the experience a bit and avoids annoying path errors.

Something like:
```
pushd $(dirname $0)
pushd $(git rev-parse --show-toplevel)
```

File: scripts/demo.with_docker.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+#
+# TODO to replace this with a properly-agnostic mount point inside the
+# container.
+#
+# TODO: turn this into a real integration test...
+
+set -eux
+
+# Get the path to this repo; we'll mount it from the container later.
+JIANT_PATH=$(readlink -f $(dirname $0))
```
Comment (iftenney) [inline scripts/demo.with_docker.sh]: need to change this so it still points to the top-level jiant/ repo. Something like:
```
# temporarily cd to jiant/
pushd $(dirname $0)
pushd $(git rev-parse --show-toplevel)

# get absolute path (I think this works)
JIANT_PATH=$(readlink -f .)
```

File: demo.with_docker.sh (hunk 1 above)
Comment (sleepinyourhat) [reply demo.with_docker.sh]: Yes, good catch. I'll revert that here.

**Comment by sleepinyourhat** [comment]
Take another look—demo.sh works, but I have no easy access to a docker environment to test your proposed solution there.

**Pull request merged**